src/main/java/com/homework18/dao/impls/ProjectMySqlDAOImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.homework18.dao.impls;
+
+import com.homework18.dao.interfaces.ProjectDao;
+import com.homework18.model.Project;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+import javax.sql.DataSource;
+import java.util.Optional;
+
+@Component("mySqlDAO")
```

Comment:
Better use annotation @Repository here.
https://stackoverflow.com/questions/6827752/whats-the-difference-between-component-repository-service-annotations-in